### ISSUE TITLE
Prometheus resource: simplify onboarding with example + expose class type

### ISSUE THREAD

**Issue by stichiboi** [dagster-io/dagster#11463]
### What's the use case?

I've had a rough day trying to understand how to use the `prometheus_resource` 😅 even though I'm familiar with Prometheus and Pushgateway.

The main issue is there are no examples on how to use it.
The second issue is that to use it you actually need to use the regular `prometheus_client` library, and then use the resource just to push the metrics.

### Ideas of implementation

First of all I would include an example of how to use the resource.
This includes a line on `prom/pushgateway`, since the first instinct is to push stuff to the regular `prom/prometheus` server.

Then I would export the `PrometheusResource` class: this way I can use it to add typings to my variables, and use the methods safely.

Lastly, I would write wrappers in the `PrometheusResource` for all the metric types, and have those wrappers automatically add the `self.registry` which is getting pushed. 

```python
# what is currently required
from prometheus_client.metrics import Summary

prometheus = context.resources.prometheus
summary = Summary("best_story", "Time to download a single article",
                      labelnames=["status_code"],
                      registry=prometheus.registry)
prometheus.push_to_gateway("hello_dagster")


# how we could simplfy
prometheus: PrometheusResource = context.resources.prometheus
summary = prometheus.Summary("best_story", "Time to download a single article",
                      labelnames=["status_code"])
prometheus.push_to_gateway("hello_dagster")
```

### Additional information

I'm free to make the PR

Docs and exposing the `PrometheusResource` class are very easy improvements and would really help.
The metrics wrapper are less necessary, but I honestly hate to see `registry=prometheus.registry` every single time I create a metric.
Is there a reason that `PrometheusResource` is creating it's own registry? I can't think of a use case... 😟 

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by Partysun** [comment]
Yep. ✋  Looking for a solid example of usage this resource and didn't find any materials, but nope, found this :)
Pls. do this PR 💯

**Comment by dridk** [comment]
Having a /metrics endpoint to work with prometheus as a pull service would be awesome
